protoc-gen-openapiv2/internal/genopenapi/types.go

```diff
@@ -158,19 +159,38 @@ type openapiParameterObject struct {
 // supported by generation tools such as swagger-codegen and go-swagger.
 // For protoc-gen-openapiv3, we'd want to add `nullable` instead.
 type schemaCore struct {
-	Type      string          `json:"type,omitempty"`
-	Format    string          `json:"format,omitempty"`
-	Ref       string          `json:"$ref,omitempty"`
-	XNullable bool            `json:"x-nullable,omitempty"`
-	Example   json.RawMessage `json:"example,omitempty"`
+	Type      string     `json:"type,omitempty" yaml:"type,omitempty"`
+	Format    string     `json:"format,omitempty" yaml:"format,omitempty"`
+	Ref       string     `json:"$ref,omitempty" yaml:"$ref,omitempty"`
+	XNullable bool       `json:"x-nullable,omitempty" yaml:"x-nullable,omitempty"`
+	Example   RawExample `json:"example,omitempty" yaml:"example,omitempty"`
 
-	Items *openapiItemsObject `json:"items,omitempty"`
+	Items *openapiItemsObject `json:"items,omitempty" yaml:"items,omitempty"`
 
 	// If the item is an enumeration include a list of all the *NAMES* of the
 	// enum values.  I'm not sure how well this will work but assuming all enums
 	// start from 0 index it will be great. I don't think that is a good assumption.
-	Enum    []string `json:"enum,omitempty"`
-	Default string   `json:"default,omitempty"`
+	Enum    []string `json:"enum,omitempty" yaml:"enum,omitempty"`
+	Default string   `json:"default,omitempty" yaml:"default,omitempty"`
+}
+
+type RawExample json.RawMessage
+
+func (m RawExample) MarshalJSON() ([]byte, error) {
+	return (json.RawMessage)(m).MarshalJSON()
+}
+
+func (m *RawExample) UnmarshalJSON(data []byte) error {
+	return (*json.RawMessage)(m).UnmarshalJSON(data)
+}
+
+func (e RawExample) MarshalYAML() (interface{}, error) {
+	var data interface{}
+	if err := json.Unmarshal(e, &data); err != nil {
+		return nil, err
+	}
+
+	return data, nil
```

Comment:
What does this do currently? It looks to me like it would return a concrete struct wrapped in an interface, how does that get marshaled to YAML? I can see in the test that it seems to do what we want, but can you add a comment here explaining why it works?

Reply:
Added comment